Add docs about linearizable registers in repositories (#102050)
elastic/elasticsearch: docs/reference/snapshot-restore/repository-shared-file-system.asciidoc

If your shared file system uses NFS then ensure that every node is running with
the same numeric UID and GID, or else update your NFS configuration to account
for the variance in numeric IDs across nodes.

[[repository-fs-linearizable-registers]]
[discrete]
==== Linearizable register implementation

> The linearizable register implementation for shared filesystem repositories is
> based around file locking. To perform a compare-and-exchange operation on a
> register, {es} first obtains a lock on the underlying file, then writes the
> updated contents under the same lock, which ensures that the file has not
> changed in the meantime.

Reviewer: ```suggestion
The linearizable register implementation for shared filesystem repositories is
based around file locking. To perform a compare-and-exchange operation on a
register, {es} first locks the underlying file and then writes the
updated contents under the same lock. This ensures that the file has not
changed in the meantime.
```
